router/api.go

```diff
@@ -0,0 +1,199 @@
+	Instance().Terminate()
+	db, err := dbservice.ConfigureDb(dbPath, "", "")
+	if err != nil {
+		return err
+	}
+	Instance().SetBolt(db)
+	Instance().cleanChannels(true)
+	return nil
+}
+
+func WithDefaultConfigAndDbPath(dbPath string) error {
+	return WithFileConfigAndDbPath(defaultConfigPath, dbPath)
+}
+
+func WithFileConfigAndDbPath(cfgPath, dbPath string) error {
+	Instance().Terminate()
+	return Instance().ApplyFileCfg(cfgPath, "", dbPath, true)
+}
+
+func AquaServerUrl(aquaServerUrl string) { //optional
+	Instance().setAquaServerUrl(aquaServerUrl)
+}
+
+func DBMaxSize(dbMaxSize int) { //optional
+}
+
+func DBTestInterval(dbTestInterval int) { //optional
+}
+
+func DBRemoveOldData(dbRemoveOldData int) { //optional
+}
+
+//------------------Outputs-------------------
+func AddOutput(output *data.OutputSettings) error {
+	return Instance().addOutput(output)
+}
+func UpdateOutput(output *data.OutputSettings) error {
+	err := Instance().deleteOutput(output.Name, false)
+	if err != nil {
+		return err
+	}
+	return Instance().addOutput(output)
+}
+func ListOutputs() []data.OutputSettings {
+	return Instance().listOutputs()
+}
+
+func DeleteOutput(name string) error {
+	return Instance().deleteOutput(name, true)
+}
+
+//-----------------------------------------------
+
+//------------------Routes--------------------
+func AddRoute(route *routes.InputRoute) {
+	Instance().addRoute(route)
+}
+
+func DeleteRoute(name string) error {
+	return Instance().deleteRoute(name)
+}
+func ListRoutes() []routes.InputRoute {
+	return Instance().listRoutes()
+}
+func UpdateRoute(route *routes.InputRoute) error {
+	err := Instance().deleteRoute(route.Name)
+	if err != nil {
+		return err
+	}
+	Instance().addRoute(route)
+	return nil
+}
+
+//-----------------------------------------------
+
+//-------------------Templates-------------------
+func AddTemplate(template *data.Template) error {
+	return Instance().addTemplate(template)
+}
+
+//helper method
+func AddRegoTemplateFromFile(name, filename string) error {
+	b, err := os.ReadFile(filename)
+
+	if err != nil {
+		return err
+	}
+
+	return AddTemplate(&data.Template{
+		Name: name,
+		Body: string(b),
+	})
+
+}
+
+func UpdateTemplate(template *data.Template) error {
+	err := Instance().deleteTemplate(template.Name, true)
+
+	if err != nil {
+		return err
+	}
+
+	return Instance().addTemplate(template)
+}
+
+func DeleteTemplate(name string) error {
+	return Instance().deleteTemplate(name, true)
+}
+
+func ListTemplates() []string {
+	/*
+		There is nothing to update (as only one property defines template).
+		So only list of template names returned
+	*/
+	templates := Instance().templates
+	names := make([]string, 0, len(templates))
+	for n := range templates {
+		names = append(names, n)
+	}
+	return names
+}
+
+//-----------------------------------------------
+
+func Send(b []byte) {
+	//Instance().Send(b)
+	Instance().handle(bytes.ReplaceAll(b, []byte{'`'}, []byte{'\''}))
+}
```

Comment:
What does this function do?

Reply:
it calls handler directly. no queue is involved